### PR TITLE
Section III mapped for F01 through F21

### PR DESCRIPTION
**Pull request by ColinMaudry**
The main difficulty was with the field `/LEFTI/PARTICULAR_PROFESSION` that has slightly different modelling rules depending on the form. Sometimes its a boolean (F01, type="services"), sometimes it's a string (F12, type="text_ft_multi_lines").

The XML Schema doesn't help much. I'd be grateful if could have a look at it.

Didn't map for:

- defense forms
- T01 and T02 forms

### CODE REVIEW

File: output/mapping/F01_2014.csv [hunk 1]
```diff
@@ -87,19 +87,19 @@ xpath,label-key,index,comment,guidance
...
-/LEFTI/ECONOMIC_FINANCIAL_MIN_LEVEL,min_standards_required,,,""
-/LEFTI/TECHNICAL_CRITERIA_DOC,criteria_selection_docs,,,""
-/LEFTI/TECHNICAL_PROFESSIONAL_INFO,info_evaluating_weth_requir,,,""
-/LEFTI/TECHNICAL_PROFESSIONAL_MIN_LEVEL,min_standards_required,,,""
-/LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
-/LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/PARTICULAR_PROFESSION,particular_profession_reserved,,,""
-/LEFTI/REFERENCE_TO_LAW,ref_law_reg_prov,,,""
+/LEFTI/SUITABILITY,info_evaluating_requir,,,"Map to `tender.participationConditions.pursuingProfessionalActivity`"
+/LEFTI/ECONOMIC_CRITERIA_DOC,criteria_selection_docs,,,"Set `tender.participationConditions.hasEconomicCriteriaInDocuments` to `true`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: Done.

File: output/mapping/F12_2014.csv [hunk 2]
```diff
@@ -52,8 +52,8 @@ xpath,label-key,index,comment,guidance
...
 /LEFTI,info_legal,III,,"The standard forms express exclusion grounds and selection criteria in this section as unstructured text. In the [introductory note](https://github.com/eForms/eForms/raw/master/20180604_eForms_consultation_introduction.docx) to its [eForms consultation](https://github.com/eForms/eForms), the European Commission proposes to create an extension to eForms to implement the [European Single Procurement Document (ESPD)](http://ec.europa.eu/growth/single-market/public-procurement/e-procurement/espd/), which expresses this information as structured data, following the model of the [Core Criterion and Core Evidence Vocabulary](https://joinup.ec.europa.eu/solution/core-criterion-and-core-evidence-vocabulary). With respect to OCDS, work on this section is postponed until either the extension to eForms is available or until the [ESPD Exchange Data Model](https://github.com/ESPD/ESPD-EDM) is stable."
 /LEFTI/CRITERIA_SELECTION,criteria_selection,III.1.10,,""
-/LEFTI/PARTICULAR_PROFESSION,dc_indicate_profession,,,""
-/LEFTI/NO_PARTICULAR_PROFESSION,_no,,,""
+/LEFTI/PARTICULAR_PROFESSION,dc_indicate_profession,,,"Map to `tender.contractTerms.reservedToProfession`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F12_2014.csv]: You mean mapping both to `.reservedToProfessionLawReference`? They don't have the same semantics: expect name(s) of profession(s) while the other expect a reference to a legal text.

I would keep them as distinct fields.

File: output/mapping/F01_2014.csv [hunk 3]
```diff
@@ -87,19 +87,19 @@ xpath,label-key,index,comment,guidance
...
-/LEFTI/ECONOMIC_FINANCIAL_INFO,info_evaluating_weth_requir,,,""
-/LEFTI/ECONOMIC_FINANCIAL_MIN_LEVEL,min_standards_required,,,""
-/LEFTI/TECHNICAL_CRITERIA_DOC,criteria_selection_docs,,,""
-/LEFTI/TECHNICAL_PROFESSIONAL_INFO,info_evaluating_weth_requir,,,""
-/LEFTI/TECHNICAL_PROFESSIONAL_MIN_LEVEL,min_standards_required,,,""
-/LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
-/LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/PARTICULAR_PROFESSION,particular_profession_reserved,,,""
-/LEFTI/REFERENCE_TO_LAW,ref_law_reg_prov,,,""
+/LEFTI/SUITABILITY,info_evaluating_requir,,,"Map to `tender.participationConditions.pursuingProfessionalActivity`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: Fixed by changing structure and use codes instead.

File: output/mapping/F01_2014.csv [hunk 4]
```diff
@@ -87,19 +87,19 @@ xpath,label-key,index,comment,guidance
...
+/LEFTI/ECONOMIC_CRITERIA_DOC,criteria_selection_docs,,,"Set `tender.participationConditions.hasEconomicCriteriaInDocuments` to `true`"
+/LEFTI/ECONOMIC_FINANCIAL_INFO,info_evaluating_weth_requir,,,"Map to `tender.participationConditions.economicSelectionCritera`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: Fixed.

File: output/mapping/F01_2014.csv [hunk 5]
```diff
@@ -87,19 +87,19 @@ xpath,label-key,index,comment,guidance
...
+/LEFTI/ECONOMIC_FINANCIAL_MIN_LEVEL,min_standards_required,,,"Map to `tender.participationConditions.economicRequiredStandards`"
+/LEFTI/TECHNICAL_CRITERIA_DOC,criteria_selection_docs,,,"Set `tender.participationConditions.hasTechnicalCriteriaInDocuments` to `true`"
+/LEFTI/TECHNICAL_PROFESSIONAL_INFO,info_evaluating_weth_requir,,,"Map to `tender.participationConditions.technicalSelectionCritera`"
+/LEFTI/TECHNICAL_PROFESSIONAL_MIN_LEVEL,min_standards_required,,,"Map to `tender.participationConditions.technicalRequiredStandards`"
+/LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,"Set `tender.participationConditions.hasReservedParticipation` to `true`"
+/LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.hasReservedExecution` to `true`
+/LEFTI/PARTICULAR_PROFESSION,particular_profession_reserved,,,Set `tender.contractTerms.hasReservedToProfession` to `true`
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: Fixed

File: output/mapping/F07_2014.csv [hunk 6]
```diff
@@ -66,9 +66,9 @@ xpath,label-key,index,comment,guidance
...
 /LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
 /LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/QUALIFICATION,qs_qualification,III.1.9,,""
-/LEFTI/QUALIFICATION/CONDITIONS,qs_conditions_qualify,,,""
-/LEFTI/QUALIFICATION/METHODS,qs_methods_verified,,,""
+/LEFTI/QUALIFICATION,qs_qualification,III.1.9,,Discard
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F07_2014.csv]: Fixed

File: output/mapping/F01_2014.csv [hunk 7]
```diff
@@ -87,19 +87,19 @@ xpath,label-key,index,comment,guidance
...
+/LEFTI/REFERENCE_TO_LAW,ref_law_reg_prov,,,Map to `tender.contractTerms.reservedToProfessionLawReference`
 /LEFTI/PERFORMANCE_CONDITIONS,other_conditions,III.2.2,,Map to `tender.contractTerms.termsPerformance`
-/LEFTI/PERFORMANCE_STAFF_QUALIFICATION,staff_responsible_indicate,,,""
+/LEFTI/PERFORMANCE_STAFF_QUALIFICATION,staff_responsible_indicate,,,Set `tender.contractTerms.hasObligationStaffQualificationInformation` to `true`
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: And it probably shouldn't be in contractTerms. I had created `tender.participationConditions`, would it fit better there?

File: output/mapping/F05_2014.csv [hunk 8]
```diff
@@ -100,9 +100,9 @@ xpath,label-key,index,comment,guidance
 /LEFTI/RULES_CRITERIA,descr_brief_rules,,,""
 /LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
 /LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/DEPOSIT_GUARANTEE_REQUIRED,deposits_required,III.1.6,,""
-/LEFTI/MAIN_FINANCING_CONDITION,financing_conditions,III.1.7,,""
-/LEFTI/LEGAL_FORM,legal_form_taken,III.1.8,,""
+/LEFTI/DEPOSIT_GUARANTEE_REQUIRED,deposits_required,III.1.6,,"Map to `tender.participationConditions.depositsGuarantees`"
+/LEFTI/MAIN_FINANCING_CONDITION,financing_conditions,III.1.7,,Map to `tender.participationConditions.financingConditions`
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Isn't it better phrased with "Financial terms"?

[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: I will move the term to `.contractTerms` to match eForms structure.

[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Wouldn't "Financial terms" be better phrased?
I will move the term to `.contractTerms` to match eForms structure.

File: output/mapping/F21_2014.csv [hunk 9]
```diff
@@ -83,7 +83,7 @@ xpath,label-key,index,comment,guidance
 /LEFTI/RULES_CRITERIA,descr_brief_rules,,,""
 /LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
 /LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/RESERVED_ORGANISATIONS_SERVICE_MISSION,reserved_public_mission,,,""
+/LEFTI/RESERVED_ORGANISATIONS_SERVICE_MISSION,reserved_public_mission,,,"Set `tender.participationConditions.hasReservedParticipationPublicService` to `true`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F21_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/07a8ce3be341b796e8818299f7801337241fa2e5

File: output/mapping/F12_2014.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F12_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/55139fd2cc7bdcdfd82f9dcb38c404dfd248c562

File: output/mapping/F01_2014.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F01_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/8450a6171ddab6a135e7c36936c71ec991c7538d

File: output/mapping/F05_2014.csv [hunk 10]
```diff
@@ -100,9 +100,9 @@ xpath,label-key,index,comment,guidance
...
+/LEFTI/LEGAL_FORM,legal_form_taken,III.1.8,,"Map to `tender.participationConditions.groupLegalForm`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/e11bf2473c830e9159802b6886cdad00e0b93642

File: output/mapping/F05_2014.csv [hunk 11]
```diff
@@ -100,9 +100,9 @@ xpath,label-key,index,comment,guidance
 /LEFTI/RULES_CRITERIA,descr_brief_rules,,,""
 /LEFTI/RESTRICTED_SHELTERED_WORKSHOP,restricted_sheltered_workshop,,,""
 /LEFTI/RESTRICTED_SHELTERED_PROGRAM,restricted_sheltered_program,,,Set `tender.contractTerms.reservedExecution` to `true`
-/LEFTI/DEPOSIT_GUARANTEE_REQUIRED,deposits_required,III.1.6,,""
-/LEFTI/MAIN_FINANCING_CONDITION,financing_conditions,III.1.7,,""
-/LEFTI/LEGAL_FORM,legal_form_taken,III.1.8,,""
+/LEFTI/DEPOSIT_GUARANTEE_REQUIRED,deposits_required,III.1.6,,"Map to `tender.participationConditions.depositsGuarantees`"
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Yes, but it's a boolean in eForms, whereas it's a string in TED. Let's mint both, a boolean and a `guaranteeRequired.description`. And I'll move both to `.submissionTerms`.

[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/90d07ce9a31dc7eb86529db91b46dbb8d4a877bd

File: output/mapping/F05_2014.csv (hunk 8 above)
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply output/mapping/F05_2014.csv]: Done in https://github.com/open-contracting/european-union-support/pull/15/commits/90d07ce9a31dc7eb86529db91b46dbb8d4a877bd

**Comment by ColinMaudry** [comment]
I think I have implemented all the changes.

File: output/mapping/F01_2014.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (jpmckinney) [reply output/mapping/F01_2014.csv]: I'm not 100% sure to what this relates. The form has it under "Conditions related to the contract", so I assume it's deliberately not under "Conditions for participation". eForms puts it under "Other requirements"; there is no relevant discussion on GitHub of its semantics or model. ESPD doesn't seem to include it…

Article 19 (Economic operators) of 2014/24/EU is directly relevant, which makes it seem more like a submission term:

> legal persons may be required to indicate, in the tender or the request to participate, the names and relevant professional qualifications of the staff to be responsible for the performance of the contract in question.

I've asked Jachym via email, but for now I'll move it to Submission Terms.

[the remark this answers is not in the document]
Comment (jpmckinney) [reply output/mapping/F01_2014.csv]: To be conservative, I'll actually move it to Other Requirements for now.

**Comment by jpmckinney** [comment]
Quite a few tweaks:

Field name changes:

* Fix `type` to `documentType`
* Rename `minimumLevel` to `minimum`, to be more flexible for future criteria, which might set minimum values, percentages, ratios, etc.
* Rename `participationConditions` to `otherRequirements`
  * As mentioned by Giampaollo Sellitto in https://github.com/eForms/eForms/issues/296#issuecomment-472861002, "participation criteria" or "participation conditions" are easy to confuse with "selection criteria" from a legal perspective. This is the reason ESPD uses "other criteria" and eForms uses "other requirements".
* Rename `reservedTo` to `reservedParticipation`, mirroring `hasReservedExecution`
* Rename `termsPerformance` to `performanceTerms`, mirroring `financialTerms`
* Rename `hasPerformingStaffQualification` to `requiresStaffNamesQualifications`, for clarity
* Rename `objectiveRulesCriteria` to `reductionCriteria`, see 2014/24/EU Article 65
* Rename `hasGuaranteeRequired` to `hasRequiredGuarantees`, to put words in regular order, and to allow multiple guarantees (like in some countries) https://github.com/open-contracting/standard/issues/900

Modelling changes:

* Change `selectionCriteria` to an object, with a `criteria` array as a sub-field, like with `awardCriteria`
* Remove `isReservedToProfession` and rename `reservedToProfession` to `reservedExecution`, and use `hasReservedExecution` and `reservedExecution` for both sheltered employment programmes and professions
* Move `hasRequiredGuarantees` from `otherRequirements` to `submissionTerms`, see eForms
* Move `tendererLegalForm` from `otherRequirements` to `contractTerms`, see eForms

Code changes:

* Rename 'economicFinancial' to 'economic', to be consistent with the new `documentType.csv` code
* Rename 'economicCriteriaSelection' to 'economicSelectionCriteria', to put words in regular order
* Rename 'technicalCriteriaSelection' to 'technicalSelectionCriteria', to put words in regular order
* Rename 'publicServiceMission' to 'publicServiceMissionOrganization' and 'shelteredWorkshops' to 'shelteredWorkshop', so that codes describe a singular organization type

All forms:

* Copy mappings across forms

Other changes not related to this pull request:

* Rename `maximumNumberParticipants` to `maximumParticipants`
* Rename `durationRationale` to `periodRationale`
* Fix `electronicInvoicing` to `electronicInvoicingPolicy`
* Fix `electronicCatalogue` to `electronicCataloguePolicy`
* Use "nearest earlier `date`" construction
* Fix single quotes instead of backticks for code
* Remove trailing whitespace

**Comment by ColinMaudry** [comment]
Wow, you had a lot of editing to do on top of my work.I'll try to pay more attention to eForms structure and wording.

**Pull request merged**